python/graphstorm/model/lp_gnn.py

```diff
@@ -205,12 +231,27 @@ def run_lp_mini_batch_predict(decoder, emb, loader, device):
                 # We do not concatenate rankings into a single
                 # ranking tensor to avoid unnecessary data copy.
                 pos_score, neg_score = s
-                if canonical_etype in ranking:
-                    ranking[canonical_etype].append(calc_ranking(pos_score, neg_score))
-                else:
-                    ranking[canonical_etype] = [calc_ranking(pos_score, neg_score)]
-
-        rankings = {}
+                assert pos_score.shape[0] == neg_score.shape[0], \
+                    "There should be as many negative lists as there are positive examples"
+                score_ranking = calc_ranking(pos_score, neg_score)
+                ranking[canonical_etype].append(score_ranking)
+                # Set the number of candidates for each positive example
+                # (equal to neg_score.shape[0])
+                # which will be the number of negatives (equal to neg_score.shape[1])
+                # plus one for the positive example
+                lengths_tensor = th.tensor(neg_score.shape[0] * [neg_score.shape[1] + 1])
```

Comment:
Should it be a single value as (neg_score.shape[1] + 1)? 
Why do you compute the total number of negatives in a batch?

Reply:
We need one length per positive, we are using `*` as a way to create a list of all same numbers, e.g. `3 * [4] => [4, 4, 4]`.

So the tensor will just have the length of the examples with all same values.

Reply:
I c.